src/builtins.c

```diff
@@ -30,6 +30,9 @@
 extern "C" {
 #endif
 
+// htable to reduce objectid execution time
+htable_t object_table;
```

Comment:
You need to make this local to the `jl_object_id` function, initialize it with `htable_new(&object_table, 0)`, and finalize it with `htable_free(&object_table)`

Reply:
You still need to make this a local variable / argument to all of the relevant functions

Reply:
Since I made `object_table` local, I had to introduce them as arguments to functions like `immut_id_`, `type_object_id_`, `jl_object_id` and `jl_object_id__cold`. However, `jl_object_id_` is a function shared across different libraries, and therefore, I had to change it's function definition. After the last change, when I tried running `make`, it gave a segfault, and mentioned that the error was due to `binaryplatforms.jl`. I tried running `make cleanall` and `make`, but the error still persisted.

Reply:
The full trace after running `make debug` is as follows:
```
signal (11): Segmentation fault
in expression starting at binaryplatforms.jl:422
ptrhash_lookup_bp_r at /home/samarth/julia/src/support/ptrhash.c:26
ptrhash_bp_r at /home/samarth/julia/src/support/ptrhash.c:26
ptrhash_bp at /home/samarth/julia/src/support/ptrhash.c:26
ijl_object_id_ at /home/samarth/julia/src/builtins.c:426
objectid at ./reflection.jl:340 [inlined]
hash at ./hashing.jl:25 [inlined]
hash at ./hashing.jl:20 [inlined]
hashindex at ./dict.jl:169 [inlined]
ht_keyindex2! at ./dict.jl:310
setindex! at ./dict.jl:384
grow_to! at ./dict.jl:159
grow_to! at ./dict.jl:145 [inlined]
dict_with_eltype at ./abstractdict.jl:578 [inlined]
Dict at ./dict.jl:129
unknown function (ip: 0x7f11060c0062)
_jl_invoke at /home/samarth/julia/src/gf.c:2367
ijl_apply_generic at /home/samarth/julia/src/gf.c:2549
Dict at ./dict.jl:125
jl_fptr_args at /home/samarth/julia/src/gf.c:2128
_jl_invoke at /home/samarth/julia/src/gf.c:2367
ijl_apply_generic at /home/samarth/julia/src/gf.c:2549
jl_apply at /home/samarth/julia/src/julia.h:1827
do_call at /home/samarth/julia/src/interpreter.c:126
eval_value at /home/samarth/julia/src/interpreter.c:215
eval_stmt_value at /home/samarth/julia/src/interpreter.c:166
eval_body at /home/samarth/julia/src/interpreter.c:594
jl_interpret_toplevel_thunk at /home/samarth/julia/src/interpreter.c:750
jl_toplevel_eval_flex at /home/samarth/julia/src/toplevel.c:906
jl_eval_module_expr at /home/samarth/julia/src/toplevel.c:203
jl_toplevel_eval_flex at /home/samarth/julia/src/toplevel.c:709
jl_parse_eval_all at /home/samarth/julia/src/toplevel.c:1040
ijl_load_ at /home/samarth/julia/src/toplevel.c:1087
include at ./boot.jl:366 [inlined]
include at ./Base.jl:315
include at ./Base.jl:17
unknown function (ip: 0x7f11060a3ac2)
_jl_invoke at /home/samarth/julia/src/gf.c:2348
ijl_apply_generic at /home/samarth/julia/src/gf.c:2549
jl_apply at /home/samarth/julia/src/julia.h:1827
do_call at /home/samarth/julia/src/interpreter.c:126
eval_value at /home/samarth/julia/src/interpreter.c:215
eval_stmt_value at /home/samarth/julia/src/interpreter.c:166
eval_body at /home/samarth/julia/src/interpreter.c:594
jl_interpret_toplevel_thunk at /home/samarth/julia/src/interpreter.c:750
jl_toplevel_eval_flex at /home/samarth/julia/src/toplevel.c:906
jl_eval_module_expr at /home/samarth/julia/src/toplevel.c:203
jl_toplevel_eval_flex at /home/samarth/julia/src/toplevel.c:709
jl_parse_eval_all at /home/samarth/julia/src/toplevel.c:1040
ijl_load_ at /home/samarth/julia/src/toplevel.c:1087
include at ./boot.jl:366
unknown function (ip: 0x7f110a71bee6)
_jl_invoke at /home/samarth/julia/src/gf.c:2367
ijl_apply_generic at /home/samarth/julia/src/gf.c:2549
jl_apply at /home/samarth/julia/src/julia.h:1827
do_call at /home/samarth/julia/src/interpreter.c:126
eval_value at /home/samarth/julia/src/interpreter.c:215
eval_stmt_value at /home/samarth/julia/src/interpreter.c:166
eval_body at /home/samarth/julia/src/interpreter.c:594
jl_interpret_toplevel_thunk at /home/samarth/julia/src/interpreter.c:750
jl_toplevel_eval_flex at /home/samarth/julia/src/toplevel.c:906
jl_parse_eval_all at /home/samarth/julia/src/toplevel.c:1040
ijl_load_ at /home/samarth/julia/src/toplevel.c:1087
ijl_load at /home/samarth/julia/src/toplevel.c:1100
exec_program at /home/samarth/julia/src/jlapi.c:521
true_main at /home/samarth/julia/src/jlapi.c:574
jl_repl_entrypoint at /home/samarth/julia/src/jlapi.c:706
jl_load_repl at /home/samarth/julia/cli/loader_lib.c:271
main at /home/samarth/julia/cli/loader_exe.c:59
__libc_start_main at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
_start at /home/samarth/julia/usr/bin/julia-debug (unknown line)
Allocations: 23034101 (Pool: 23026560; Big: 7541); GC: 25
Segmentation fault (core dumped)
```